src/python/pants/engine/internals/graph_test.py

```diff
@@ -1072,12 +1084,15 @@ def test_parametrize_multi(generated_targets_rule_runner: RuleRunner) -> None:
     )
 
 
+@pytest.mark.xfail(
+    reason="Add support for parametrize and `overrides`. When doing so, test the interaction "
+    "of overridding a field in the generator, then overridding that same field."
+)
```

Comment:
I'm not sure that it will be possible to implement overrides correctly after already having expanded parameters at the `BUILD` file parsing level. I had started out implementing #14408 this way (part of the reason why it and #14377 ended up so large!), but the challenge is that an override can prevent parametrization from occurring... and if it has already happened at `BUILD` parse time, it's too late to undo it.

As an example, the definition:
```python
python_sources(
    ..,
    resolve=parametrize('a', 'b'),
    overrides={'source1.py': {'resolve': "c"}},
)
```
...should result in only one copy of the target for `source1.py`. But if parametrization is applied at `BUILD` parse time, then at target generation time, there are two generator targets. That has two problems:
1.  Each of the generators would try to generate a target for `source1.py`, causing a collision
2. The generated target for `source1.py` would need to _remove_ parametrization from the address of the generator target, while generating since the `resolve` field was no longer `parametrized`.